### DISCUSSION TITLE
Slow Type Check (Result in slow compile time)

### DISCUSSION THREAD

**Discussion by jeffersonsetiawan** [pointfreeco/swift-prelude#159] [Q&A]
I loved Prelude and used it a lot in my projects.

But recently, one of member in my team found out that if we use a lot of it, it cause slow type check (he used 8 pipe) like this
```
let a8: TestingOnly = TestingOnly(a: true)
            |> \.a .~ false
            |> \.a .~ false
            |> \.a .~ false
            |> \.a .~ false
            |> \.a .~ false
            |> \.a .~ false
            |> \.a .~ false
            |> \.a .~ false
```
and the type check can be as slow as 10s.

![slow](https://user-images.githubusercontent.com/14871122/117656050-0aece080-b1c2-11eb-838c-30c1a1b2d494.jpg)

For temporary solution I suggest my team not to use lot of pipe chaining to have faster type checking or breaking it to smaller chunks

Demo project: https://github.com/jeffersonsetiawan/SlowTypeCheckLens

Really appreciate if someone can make this faster 🙏🏻

**Comment by stephencelis** [comment]
Sorry for the radio silence in this repo. It's more of an experimental one compared to our more publicly maintained ones!

This is unfortunately a limitation of the Swift compiler and not something we can solve at the library-level. Our "fix" is the same as the one you suggest (breaking the expression down into smaller ones).

Going to migrate this to a discussion for now, thanks!
